frontend/src/routes/index.lazy.tsx

```diff
@@ -1,33 +1,13 @@
 import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
 import { createLazyFileRoute, useNavigate } from "@tanstack/react-router";
+import { withAuthRedirect } from "~/components/helper/auth-redirect";
 import { ApiResponse, apiClient } from "../api/client";
 import { deleteCookie, getCookie } from "../utils";
 
 export const Route = createLazyFileRoute("/")({
-  component: Index
+  component: withAuthRedirect(AuthedView)
 });
 
-function Index() {
-  const navigate = useNavigate({ from: "/" });
-
-  const query = useQuery({
-    queryKey: ["AUTHED_USER"],
-    queryFn: ({ signal }) => {
-      return apiClient.GET("/api/auth/me/", { signal });
-    }
-  });
-  if (query.isLoading) {
-    return <div className="text-3xl font-bold">Loading... with tailwind</div>;
-  }
-
-  const authedUser = query.data?.data?.user;
-  if (!authedUser) {
-    navigate({ to: "/login" });
-    return null;
-  }
-  return <AuthedView user={authedUser} />;
-}
-
 function AuthedView({ user }: ApiResponse<"get", "/api/auth/me/">) {
```

Comment:
`user` here will be undefined because this view doesn't receive the user as a prop. You could instead use `useQuery(['AUTHED_USER'], {...})` (the same code from the auth-redirect. 

I suggest you to refactor that hook into its own file/function, like `useAuthedUser`